### PR TITLE
Update QR formatting

### PR DESCRIPTION
**Pull request by AleksandrPanov**
Added new file log and new formatting
python args:
`-p=qrcodes\detection -a=0.3 --metric=intersection_over_union -alg=opencv_aruco`

new_bench

all_qr

### CODE REVIEW

File: python_benchmarks/qr_codes/qr.py [hunk 1]
```diff
@@ -4,21 +4,75 @@
...
+    if metric is TypeNorm.intersection_over_union:
+        return 1.
+    raise TypeError("this TypeNorm isn't supported")
+
+
+def get_norm(gold_corner, corner, metric):
+    if metric is TypeNorm.l1:
+        return LA.norm((gold_corner - corner).flatten(), 1)
+    if metric is TypeNorm.l2 or metric is TypeNorm.intersection_over_union:
+        return LA.norm((gold_corner - corner).flatten(), 2)
```
Comment (asmorkalov) [inline python_benchmarks/qr_codes/qr.py]: I propose to use name for parameter `ord`. Current `1` and `2`, etc is not obvious for reader.

**Review by asmorkalov — COMMENTED** [review]
I propose to report all metrics we have to report table, not only one.

File: python_benchmarks/qr_codes/qr.py [hunk 2]
```diff
@@ -106,60 +207,85 @@ def get_gold_corners(label_path):
...
+    plt.xlabel('error')
+    plt.xticks(np.arange(0., float(accuracy) + .25, .25))
+    plt.ylabel('frequency')
+    # plt.show()
+    plt.savefig(path + '/' + category + '_' + obj_type + '.jpg')
+    plt.close()
+    return category_statistic
+
+
+def print_statistics(distances, accuracy, output_path, per_image_statistic):
```
Comment (asmorkalov) [inline python_benchmarks/qr_codes/qr.py]: I propose to rename `accuracy` to `accuracy_threshold` everywhere, including command line parameters. It makes code analysis dramatically simpler.

File: python_benchmarks/qr_codes/qr.py [hunk 3]
```diff
@@ -29,25 +83,63 @@ def __init__(self):
         self.decoded_info = []
         self.detector = None
 
+    def detect_and_decode(self, image):
+        return False, self.decoded_info, self.detected_corners
+
+    def detect_and_check(self, image, gold_corners, accuracy, metric, show_detected=False):
```
Comment (asmorkalov) [inline python_benchmarks/qr_codes/qr.py]: the same for `accuracy` field.

File: python_benchmarks/qr_codes/qr.py (hunk 1 above)
Comment (AleksandrPanov) [reply python_benchmarks/qr_codes/qr.py]: added

**Comment by asmorkalov** [comment]
Output path works incorrectly. E.g.:
```
python3 ./qr.py -alg opencv -o opencv_classic_4.x -p /home/alexander/Projects/OpenCV/opencv_benchmarks/python_benchmarks/qrcodes_v3_dataset/detection/
...
Traceback (most recent call last):
  File "./qr.py", line 379, in <module>
    main()
  File "./qr.py", line 362, in main
    dump_log(filename, img_name, output, nearest_distance_from_gold, decoded)
  File "./qr.py", line 297, in dump_log
    os.mkdir(output_dict)
FileNotFoundError: [Errno 2] No such file or directory: 'opencv_classic_4.x/report_2024_02_15-09_10_04_AM'
```

**Comment by asmorkalov** [comment]
I would say, that `report_2024_02_15-09_10_04_AM` should be default value, if output path is not set.

**Comment by asmorkalov** [comment]
Need to tune generated diagram, it's not readable.
_blurred_qr

**Comment by asmorkalov** [comment]
What distance 40 means?

File: python_benchmarks/qr_codes/qr.py [hunk 4]
```diff
@@ -106,65 +214,108 @@ def get_gold_corners(label_path):
...
+    plt.close()
+    return category_statistic
+
+
+def print_statistics(filename, distances, accuracy_threshold, output_path, per_image_statistic):
+    output_dict = output_path + '/' + filename
+    if not os.path.exists(output_dict):
+        os.mkdir(output_dict)
+    with open(output_dict + "/" + "distances" + '.json', 'w') as fp:
+        json.dump(distances, fp, cls=NumpyEncoder)
```
Comment (asmorkalov) [inline python_benchmarks/qr_codes/qr.py]: please use `indent=4` for "human readable" json. Otherwise it's single line document.

Comment (AleksandrPanov) [reply python_benchmarks/qr_codes/qr.py]: added

**Comment by AleksandrPanov** [comment]
fixed diagram and output
image

**Review by asmorkalov — APPROVED** [review]
:+1:

**Pull request merged**